packages/cadence-parser/src/index.js

```diff
@@ -0,0 +1,40 @@
+import {invariant} from "@onflow/util-invariant"
+
+/**
+ * Parses a Cadence transaction to extract function arguments.
+ *
+ * @param {string} cadenceCode - The Cadence transaction code.
+ * @returns {Array<object>} An array of objects each containing 'name' and 'type' of each argument in the transaction.
+ * Each object has the format: { name: string, type: string }. If no valid arguments are found, or if the Cadence
+ * transaction is not properly formatted, the function will return an empty array.
+ *
+ * @example
+ * const cadenceCode = 'transaction(name: String, age: Int)'
+ * parseArguments(cadenceCode)
+ * // Returns: [{ name: 'name', type: 'String' }, { name: 'age', type: 'Int' }]
+ *
+ * @example
+ * const cadenceCode = 'transaction()'
+ * parseArguments(cadenceCode)
+ * // Returns: []
+ */
+export const parseArguments = cadenceCode => {
+  const start = cadenceCode.indexOf('(') + 1
```

Comment:
I think this needs to support the location of the `transaction` or `main` declarations being at any place in the cadence transaction/script. For example, we need to support the following:

```cadence
pub struct ExampleStruct {
  pub var exampleMsg: String
  init(exampleMsg: String) {
   self.exampleMsg = exampleMsg
  }
}

pub fun main(msg: String): ExampleStruct {
  return ExampleStruct(exampleMsg: msg)
}
```

I think it might be useful to look for the `transaction` or `pub fun main` declarations, and search for argument info from there.

Reply:
Good catch